tools/docker/Dockerfile.ubuntu22.04

```diff
@@ -5,16 +5,20 @@ FROM ubuntu:22.04 as build-base
 SHELL ["/bin/bash", "-c"]
 
 ARG DEBIAN_FRONTEND=noninteractive
+
+#= Install packages ========================================================
+# Please keep the list sorted by name
 RUN apt update && apt-get install -y --no-install-recommends \
     build-essential \
     ca-certificates \
+    clang-format \
     curl \
     git-core \
+    gnupg \
     libssl-dev \
-    python-is-python3 \
     python3-pip \
-    wget \
-    gnupg
+    python-is-python3 \
+    wget
```

Comment:
I suggest adding the clang-format package in the last stage here:

https://github.com/asterinas/asterinas/blob/4eff679eacb2415b3e3077d6c0d261e742d87f23/tools/docker/Dockerfile.ubuntu22.04#L220-L241

The `build-base` phase contains only necessary packages for the container's building processes. The lesser the packages modified here the more build cache we can utilize. As for required packages to develop Asterinas, putting them in the last phase is better.

Maybe polish the comments on the `build-base` phase further, since the comments here does not reflect the design mentioned above.

Reply:
That's a good point. I didn't examine this Dockerfile closely enough to notice that there are two stages of package installation that actually serve different purposes.